dns: Improve error message on DNS interface not found

cathay4t opened the pull request:
Change the error message

from:

  libnmstate.error.NmstateValueError: Failed to find suitable
  interface for saving DNS name servers: 8.8.8.8

to:
    libnmstate.error.NmstateValueError: Failed to find suitable
    interface for saving DNS name servers: 8.8.8.8, nmstate only support
    saving DNS to interface with static gateay or auto interface with
    auto-dns:false

Unit test case included.

ffmancera commented on libnmstate/dns.py:
```diff
@@ -123,7 +123,9 @@ def gen_metadata(self, ifaces, route_state):
                     "Failed to find suitable interface for saving DNS "
-                    "name servers: %s" % server
+                    "name servers: %s, nmstate only support saving DNS to "
+                    "interface with static gateay or auto interface with "
```
There is a typo in `gateway`. Other than that, it looks good. It is wrong in the commit message too.

cathay4t replied:
Fixed. Thanks.

cathay4t committed:
dns: Improve error message on DNS interface not found

Change the error message

from:

  libnmstate.error.NmstateValueError: Failed to find suitable
  interface for saving DNS name servers: 8.8.8.8

to:
    libnmstate.error.NmstateValueError: Failed to find suitable
    interface for saving DNS name servers: 8.8.8.8, nmstate only support
    saving DNS to interface with static gateway or auto interface with
    auto-dns:false

Unit test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

ffmancera reviewed (approved):
LGTM, thanks!
